### PR TITLE
[stable/grafana] add pod annotations to grafana chart

### PR DESCRIPTION
**Pull request by willejs**
**What this PR does / why we need it**:
Currently you cannot add pod annotations to the grafana chart, only deployment annotations. Kube2iam only used pod annotations when detecting the role to use and kube2iam is very useful for the cloudwatch datasource.

**Special notes for your reviewer**:
- I would have name spaced the value `annotations` to sometthing like `deploymentAnnotations` but this is a breaking change.
- I followed the existing style of `with` instead of `if` on `.Values.podAnnotations` as per `annotations`.

@rtluckie can you review please 😸

### CODE REVIEW

**Comment by willejs** [comment]
/assign @carrodher 
Can you look at this too please?

File: stable/grafana/values.yaml [hunk 1]
```diff
@@ -17,6 +17,12 @@ downloadDashboardsImage:
   tag: latest
   pullPolicy: IfNotPresent
 
+## Pod Annotations
+# podAnnotations: {}
+
+## Deployment annotations
+# annotations: {}
```
Comment (rjkernick) [inline stable/grafana/values.yaml:24]: Can you also add these into the README.md?

**Review by rjkernick — CHANGES_REQUESTED** [review]

File: stable/grafana/values.yaml (hunk 1 above)
Comment (willejs) [reply stable/grafana/values.yaml:24]: Added, thanks.

**Review by rjkernick — APPROVED** [review]

**Pull request merged**